testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -8628,6 +8628,27 @@ void checkArrayReverseFunc(SqlOperatorFixture f0, SqlFunction function,
     f.checkNull("acos(cast(null as double))");
   }
 
+  @Test void testAcosdFunc() {
+    final SqlOperatorFixture f = fixture().withLibrary(SqlLibrary.POSTGRESQL);
+    f.setFor(SqlLibraryOperators.ACOSD, VmName.EXPAND);
+    f.checkType("acosd(0)", "DOUBLE NOT NULL");
+    f.checkType("acosd(cast(1 as float))", "DOUBLE NOT NULL");
+    f.checkType("acosd(case when false then 0.5 else null end)", "DOUBLE");
+    f.enableTypeCoercion(false)
+        .checkFails("^acosd('abc')^",
+            "Cannot apply 'ACOSD' to arguments of type "
+                + "'ACOSD\\(<CHAR\\(3\\)>\\)'\\. Supported form\\(s\\): "
+                + "'ACOSD\\(<NUMERIC>\\)'",
+            false);
+    f.checkType("acosd('abc')", "DOUBLE NOT NULL");
+    f.checkScalarApprox("acosd(0.5)", "DOUBLE NOT NULL",
+        isWithin(59.99d, 60.01d));
+    f.checkScalarApprox("acosd(cast(0.5 as decimal(3, 1)))", "DOUBLE NOT NULL",
+        isWithin(59.9d, 60.1d));
+    f.checkNull("acosd(cast(null as integer))");
+    f.checkNull("acosd(cast(null as double))");
```

Comment:
how about some tests with values outside the range -1..1?

Reply:
Added some more tests that are outside of the normal range to verify that `NaN` is returned when expected.